amd_openvx_extensions/amd_custom/source/custom_layer.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright (c) 2017 - 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "kernels.h"
+#include "custom_api.h"
+
+struct CustomLayerLocalData {
+    CustomFunctionType function;
+    customHandle custom_handle;
+    unsigned char *pCustomParameterArray;
+    customTensorDesc input_desc, output_desc;
+    void* input_mem;
+    void* output_mem;
+    unsigned int cpu_num_threads;
+    void * hipstream;
+    customBackend backend;
+};
+
+inline void Set4dTensorDesc(customTensorDesc &desc, int data_type, vx_size (&dims)[4],  vx_size (&strides)[4])
+{
```

Comment:
formatting: move the { to the previous line

Reply:
done